src/ShopBundle/objects/db/TShopOrder.class.php

```diff
@@ -540,6 +543,32 @@ protected function SaveOrderNotificationDataToOrder($bOrderSend, $oMail, $sSendT
         $this->AllowEditByAll(false);
     }
 
+    protected function SetSendOrderNotificationState(bool $orderSend): void
+    {
+        if (true === TGlobal::IsCMSMode()) {
+            return;
+        }
+
+        $aData = $this->sqlData;
+        if (true === $orderSend) {
+            $aData = $this->AddOrderNotificationDataForOrderSuccess($aData);
+        } else {
+            $aData = $this->AddOrderNotificationDataForOrderNotDone($aData);
+        }
+
+        $this->LoadFromRow($aData);
+        $this->AllowEditByAll(true);
+        $this->Save();
+        $this->AllowEditByAll(false);
+    }
+
+    protected function AddOrderNotificationDataForOrderNotDone(array $aData): array
```

Comment:
Naming: This does not "add" things (but actually only sets).

I would remove the method as it is a one-liner and thus has no usage for double code avoidance nor is it needed to better document something.

Reply:
The method is resolved.